array_api_compat/common/_helpers.py

```diff
@@ -280,13 +275,13 @@ def is_pydata_sparse_array(x: object) -> TypeIs[sparse.SparseArray]:
     if "sparse" not in sys.modules:
         return False
 
-    import sparse  # pyright: ignore[reportMissingTypeStubs]
+    import sparse
 
     # TODO: Account for other backends.
     return isinstance(x, sparse.SparseArray)
 
 
-def is_array_api_obj(x: object) -> TypeIs[_ArrayApiObj]:  # pyright: ignore[reportUnknownParameterType]
+def is_array_api_obj(x: object) -> TypeGuard[_ArrayApiObj]:
```

Comment:
Given the current definition of _ArrayApiObj, TypeIs would cause downstream failures for all unknown array api compliant libraries.

Reply:
`SupportsArrayNamespace` just states that the object has a `__array_namespace__` method, nothing more.
It's missing all the other methods and properties of an Array.
I would much rather NOT write the full Array protocol here (I did it in array-api-extra and I regret it), as this is squarely in scope for `array-api-types`.

Reply:
Today, yes, because `array-api-compat` by design does not wrap around Array classes. NumPy 1.x, Dask, CuPy, and PyTorch don't have this method even if it's mandated by the Array API standard.
`array-api-compat` shims around this by defining the `is_array_api_obj` function, which states that an Array API array is one defined by one of the four above libraries _or by any fully compliant library_.